File: Python/283.Move Zeroes.py
#把数组中的0都移动到最后，而保持其他数的顺序不变(在原数组上操作)
class Solution(object):
    def moveZeroes(self, nums):
        i , j , l = 0 , 0 , len(nums)
        while j<l:
            if nums[j]!=0:
                nums[i] , i = nums[j] , i+1
            j += 1
        while i<l:
            nums[i] , i = 0 , i+1
        return nums

s = Solution()
print(s.moveZeroes([0,1,0,3,12]))

File: C++/Two Pointers/283.Move Zeroes.cpp
/*
把数组中的0全都挪到最后。
*/

#include <iostream>
#include <vector>
#include <math.h>
using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i = 0,j = 0,length = nums.size();
        while(j < length)
        {
        	if(nums[j]!=0)
        	{
        		nums[i++] = nums[j];
        	}
        	++j;
        }
        while(i<length)
        {
        	nums[i++] = 0;
        }
    }
};

int main()
{
	Solution s;
	vector<int> nums;
	nums.push_back(2);
	nums.push_back(0);
	nums.push_back(1);
	nums.push_back(2);
	nums.push_back(0);
	nums.push_back(1);
	s.moveZeroes(nums);
	for(int i=0; i<nums.size(); ++i)
	{
		cout<<nums[i]<<endl;
	}
}